src/main/java/org/datadog/jenkins/plugins/datadog/traces/BuildSpanAction.java

```diff
@@ -10,13 +11,19 @@
  */
 public class BuildSpanAction extends InvisibleAction {
```

Comment:
Actions are usually persisted to disk, do we need to pass BuildDate around? If so let's make sure both the Action and BuildData are serializable.

Disclaimer: I'm not totally sure how Jenkins is able to persist Actions to disk if they don't implement Serializable so maybe I'm missing something. But I know that unserializable action can lead to difficult issues upon jenkins restarts.

Reply:
Good catch 👍 . Added the `Serializable` interface to those classes